### DISCUSSION TITLE
Question about weights

### DISCUSSION THREAD

**Discussion by rafaelblevin821** [BindsNET/bindsnet#631] [General]
Could anyone tell me why: 

(Sum of all weight connections) / (number of neurons) = 79 ???
eg. Summatation of all input connection weights = (55300) / n_neurons (700) = 79
eg. Summatation of all input connection weights = (63200) / n_neurons (800) = 79 
... and so on, for every value of n_neurons, the summation of the weights will always equate to: n_neurons * 79.

I am using the Diehl and Cook supervised mnist model. 

Is there a piece of code that locks the weights to a specific value per number of neurons?

**Comment by rafaelblevin821** [comment]
I figured it out for anyone wanting to know how the weights are calculated. The 79 was coming from the value that 'norm' is set to. 

So 'Norm' = (Sum of all weight connections) / (number of neurons)

**Comment by Hananel-Hazan** [comment]
While there may not be a specified reason for this normalization, it's possible that the "normal" ratio between black and white pixels in the MNIST dataset has been taken into consideration. This is just my speculation though, and perhaps someone else has a more informed explanation.
